app/presenters/check_answers/case_details_card.rb

```diff
@@ -0,0 +1,52 @@
+module CheckAnswers
+  class CaseDetailsCard < Base
+    attr_reader :case_details_form
+
+    KEY = 'case_details'.freeze
+
+    def initialize(claim)
+      @case_details_form = Steps::CaseDetailsForm.build(claim)
+    end
+
+    def route_path
+      KEY
+    end
+
+    def title
+      I18n.t('steps.check_answers.groups.about_case.case_details.title')
+    end
+
+    # TO DO: update remittal to include date of remittal when CRM457-172 is done
+    # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
+    def rows
+      [
```

Comment:
this could be streamlined:
```ruby
def rows
   row_data.map do |key, value|
      {
          key: { text: translate_table_key(KEY, key), classes: 'govuk-summary-list__value-width-50' },
          value: { text: value }
        }
   end
end

def row_data
  {
    'main_offence' => case_details_form.main_offence,
    'main_offence_date' => case_details_form.main_offence_date&.strftime('%d %B %Y'),
   ...
  }
end
```
Then rows can be part of the base class and you only need to test the row_data... this would also help simplify your tests of noise..

Reply:
this is an option but it also limits what we can do with the key of each table, sometimes I need to pass in opt to the key e.g. in additional defendants